src/v/datalake/coordinator/commit_offset_metadata.h

```diff
@@ -19,7 +19,13 @@ struct commit_offset_metadata {
     // are committed to an external catalog.
     model::offset offset;
 
-    // TODO: version? topic revision id? partition id? cluster uuid?
+    // UUID of the cluster that committed this metadata.
+    //
+    // Is std::nullopt if this was written in an older version of Redpanda that
+    // did not serialize this field.
```

Comment:
so the semantics in the case that cluster is nullopt is to then treat it like it is equal to any target uuid in order to be backwards compact?

Reply:
Right, exactly